src/transactions/TransactionFrame.cpp

```diff
@@ -66,6 +66,12 @@ TransactionFrame::getFullHash() const
 Hash const&
 TransactionFrame::getContentsHash() const
 {
+#ifdef _DEBUG
+    // force recompute
+    Hash oldHash;
+    std::swap(mContentsHash, oldHash);
```

Comment:
couple things:
* `_DEBUG` is not defined anywhere, so it's not enabled in any build by default. I was planning on maybe adding this to `extrachecks` at some point, but there are tradeoffs.
* "a bug in caching" as in we don't cache (so perf impact?)?
   * What is a lot more likely (and allowed me to find many bugs in tests) is that people mutate state even though we pre-computed the hash (ie, they forgot to call `clearCached` for example). With this change you get an assert the next time you try to access the supposedly cached hash instead of returning a stale value for the hash.

In any case I don't want to keep discussing this further in this PR - I am removing the commit that I thought was non controversial because of the dependency on `_DEBUG` and will open a different PR with that change.